src/GraphRequest.ts

```diff
@@ -367,12 +367,15 @@ export class GraphRequest {
 		let rawResponse: Response;
 		const middlewareControl = new MiddlewareControl(this._middlewareOptions);
 		this.updateRequestOptions(options);
+		const customHosts = this.config?.customHosts;
```

Comment:
I think we should verify that the custom hosts set in the configuration is valid, and if not, throw a client error instructing the developer to provide a host in the expected format. Since they may copy and paste from a URL, I wouldn't be surprised to see a `https://graph.microsoft.com/v1.0` set in the configuration.

Reply:
Added a check to throw exception if the hostname is not valid in the config